zebedee-cms/src/main/java/com/github/onsdigital/zebedee/api/Identity.java

```diff
@@ -0,0 +1,127 @@
+import javax.servlet.http.HttpServletResponse;
+import javax.ws.rs.GET;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+import static com.github.onsdigital.zebedee.logging.ZebedeeLogBuilder.logError;
+import static com.github.onsdigital.zebedee.logging.ZebedeeLogBuilder.logInfo;
+import static com.github.onsdigital.zebedee.logging.ZebedeeLogBuilder.logWarn;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static org.apache.http.HttpStatus.SC_OK;
+import static org.apache.http.HttpStatus.SC_UNAUTHORIZED;
+
+@Api
+public class Identity {
+
+	private AuthorisationService authorisationService;
+    private ServiceStore serviceStore;
+
+    static final String AUTHORIZATION_HEADER = "Authorization";
+
+    @GET
+	public void identifyUser(HttpServletRequest request, HttpServletResponse response) throws IOException,
+			UserIdentityException {
+
+		if (request.getHeader(AUTHORIZATION_HEADER) != null) {
+			ServiceAccount serviceAccount = findService(request);
+			if (serviceAccount != null) {
+				if (request.getHeader(RequestUtils.TOKEN_HEADER) != null) {
+					findUser(request, response);
+					return;
+				} else {
+					writeResponse(response, new UserIdentity(serviceAccount.getId()), SC_OK);
+				    return;
+				}
+			}
+		}
+		writeResponse(response, new Error("service not authenticated"), SC_UNAUTHORIZED);
+	}
+
+	private void findUser(HttpServletRequest request, HttpServletResponse response) throws IOException,
+			UserIdentityException {
+		String sessionID = RequestUtils.getSessionId(request);
+
+		if (StringUtils.isEmpty(sessionID)) {
+			Error responseBody = new Error("user not authenticated");
+			logWarn(responseBody.getMessage()).log();
+			writeResponse(response, responseBody, SC_UNAUTHORIZED);
+			return;
+		}
+
+		try {
+			UserIdentity identity = getAuthorisationService().identifyUser(sessionID);
+			logInfo("authenticated user identity confirmed")
+					.sessionID(sessionID)
+					.user(identity.getIdentifier())
+					.log();
+			writeResponse(response, identity, SC_OK);
+		} catch (UserIdentityException e) {
+			logError(e, "identify user failure, returning error response").log();
+			writeResponse(response, new Error(e.getMessage()), e.getResponseCode());
+		}
+	}
+
+	private ServiceAccount findService(HttpServletRequest request) throws IOException {
+        final ServiceStore serviceStore = getServiceStoreImpl();
+		String authorizationHeader = request.getHeader("Authorization");
+		if (authorizationHeader != null) {
+			authorizationHeader = authorizationHeader.toLowerCase();
+			if (authorizationHeader.toLowerCase().startsWith("bearer ")) {
+				String[] cred = authorizationHeader.split("bearer ");
+				String token = cred[1];
+				final ServiceAccount service = serviceStore.get(token);
+				if (service != null) {
+					logInfo("authenticated service account confirmed")
+							.user(service.getId())
+							.log();
+					return service;
+				} else {
```

Comment:
I think the `else` can be removed here just let the null at the end of the method return.